dpcpp/CMakeLists.txt

```diff
@@ -51,25 +51,67 @@ target_sources(ginkgo_dpcpp
     ${GKO_UNIFIED_COMMON_SOURCES}
     )
 
+## Parse the subgroup and workgroup lists
+# 1. Sort from smallest element to the biggest
+insertionSortRecursive(GINKGO_DPCPP_SUBGROUPS)
+insertionSortRecursive(GINKGO_DPCPP_WORKGROUPS)
+# 2. Extract the two biggest elements
+list(GET GINKGO_DPCPP_SUBGROUPS -1 GINKGO_DPCPP_BIGGEST_SUBGROUP)
+list(GET GINKGO_DPCPP_WORKGROUPS -1 GINKGO_DPCPP_BIGGEST_WORKGROUP)
+# 3. Combine the two lists, also format the workgroup list
+set(GINKGO_DPCPP_1D_CFG "")
+set(GINKGO_DPCPP_WORKGROUPS_FORMATTED "")
+foreach(workgroup ${GINKGO_DPCPP_WORKGROUPS})
+    foreach(subgroup ${GINKGO_DPCPP_SUBGROUPS})
+        if (GINKGO_DPCPP_1D_CFG STREQUAL "")
+            set(GINKGO_DPCPP_1D_CFG "KCFG_1D::encode(${workgroup}, ${subgroup})")
+        else()
+            set(GINKGO_DPCPP_1D_CFG "${GINKGO_DPCPP_1D_CFG}, KCFG_1D::encode(${workgroup}, ${subgroup})")
+        endif()
+    endforeach()
+    if (GINKGO_DPCPP_WORKGROUPS_FORMATTED STREQUAL "")
+        set(GINKGO_DPCPP_WORKGROUPS_FORMATTED "BlockCfg::encode(${workgroup})")
+    else()
+        set(GINKGO_DPCPP_WORKGROUPS_FORMATTED "BlockCfg::encode(${workgroup}), ${GINKGO_DPCPP_WORKGROUPS_FORMATTED}, ")
+    endif()
+endforeach()
+# 4. Reverse the subgroup list, replace ; to ,
+list(REVERSE GINKGO_DPCPP_SUBGROUPS)
+string(REPLACE ";" "," GINKGO_DPCPP_SUBGROUPS_FORMATTED "${GINKGO_DPCPP_SUBGROUPS}")
+
+## Now we can generate the DPC++ Configure file
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/base/config.hpp.in
+    ${CMAKE_CURRENT_SOURCE_DIR}/base/config.hpp @ONLY)
+
+
 ginkgo_compile_features(ginkgo_dpcpp)
 target_compile_definitions(ginkgo_dpcpp PRIVATE GKO_COMPILING_DPCPP _ONEDPL_COMPILE_KERNEL=0)
 
 set(GINKGO_DPCPP_FLAGS ${GINKGO_DPCPP_FLAGS} PARENT_SCOPE)
 target_compile_options(ginkgo_dpcpp PRIVATE "${GINKGO_DPCPP_FLAGS}")
 target_compile_options(ginkgo_dpcpp PRIVATE "${GINKGO_COMPILER_FLAGS}")
+target_compile_features(ginkgo_dpcpp PRIVATE cxx_std_17)
```

Comment:
Maybe this should be PUBLIC to avoid potential differences in code generation for header files leading to issues with ODR?